crates/rpc/rpc-eth-api/src/helpers/state.rs

```diff
@@ -269,25 +269,28 @@ pub trait LoadState: EthApiTypes {
         Self: SpawnBlocking,
     {
         self.spawn_blocking_io(move |this| {
+            let state = this.state_at_block_id_or_latest(block_id)?;
+            let base_nonce = state
+                .account_nonce(address)
+                .map_err(Self::Error::from_eth_err)?
+                .unwrap_or_default();
+
             if block_id == Some(BlockId::pending()) {
                 let address_txs = this.pool().get_transactions_by_sender(address);
-                if let Some(highest_nonce) =
-                    address_txs.iter().map(|item| item.transaction.nonce()).max()
-                {
-                    let tx_count = highest_nonce.checked_add(1).ok_or(Self::Error::from(
+                let highest_pool_nonce = address_txs
+                    .iter()
+                    .map(|item| item.transaction.nonce())
+                    .max()
+                    .unwrap_or(base_nonce);
+
+                let tx_count =
+                    highest_pool_nonce.max(base_nonce).checked_add(1).ok_or(Self::Error::from(
```

Comment:
this can now return incorrect values if there are no txs in the pool.

so the checked_add needs to be applied to the `max` nonce above